### ISSUE TITLE
Phone Pe, BHIM Not show in Android 10

### ISSUE THREAD

**Issue by KrishnaSoftwareSolution** [PatilShreyas/EasyUpiPayment-Android#89]
After integrating in Anroid 10, only Google Pay shows in payment dialog.
Other pay upi not showing

**Comment by PatilShreyas** [comment]
Maybe the apps which you mentioned are not frequently used. Try opening them and again open with UPI library.

**Comment by KrishnaSoftwareSolution** [comment]
Hello there,
All apps are frequently used.
And also done opening and closing, still only Google Pay shows.

**Comment by KrishnaSoftwareSolution** [comment]
Some security changes in Android 10
With direct invoking using package name is also showing invalid package name for phone pe

**Comment by PatilShreyas** [comment]
Thanks for reporting this. Will try to fix it soon

**Comment by KrishnaSoftwareSolution** [comment]
> Thanks for reporting this. Will try to fix it soon

Did you find problem ?
Why it is not invoking using package name itself ?

**Comment by KrishnaSoftwareSolution** [comment]
Did you find any solution ?

**Comment by AtulYadav01** [comment]
Did anyone find any solution?

**Comment by PatilShreyas** [comment]
Thanks for reporting this issue. Looks like this is happening for devices having Android 11. Currently, I don't have any device having Android 11 to test this change. I've tried to fix this change and attached APK here 👇🏻.

[***Download Sample APK***](https://github.com/PatilShreyas/EasyUpiPayment-Android/raw/v3.0.3-dev/app/app-debug.apk)

It's my humble request to test the attached APK and please let me know if it works as expected. If it works, I'll be happy to release the new version `v3.0.3`.

**Comment by AtulYadav01** [comment]
> Thanks for reporting this issue. Looks like this is happening for devices having Android 11. Currently, I don't have any device having Android 11 to test this change. I've tried to fix this change and attached APK here 👇🏻.
> 
> [_**Download Sample APK**_](https://github.com/PatilShreyas/EasyUpiPayment-Android/raw/v3.0.3-dev/app/app-debug.apk)
> 
> It's my humble request to test the attached APK and please let me know if it works as expected. If it works, I'll be happy to release the new version `v3.0.3`.

Itill facing the same issue. I tested it on MI A3.

> Thanks for reporting this issue. Looks like this is happening for devices having Android 11. Currently, I don't have any device having Android 11 to test this change. I've tried to fix this change and attached APK here 👇🏻.
> 
> [_**Download Sample APK**_](https://github.com/PatilShreyas/EasyUpiPayment-Android/raw/v3.0.3-dev/app/app-debug.apk)
> 
> It's my humble request to test the attached APK and please let me know if it works as expected. If it works, I'll be happy to release the new version `v3.0.3`.

Still facing the same issue, i tried it on MI A3.

**Comment by AtulYadav01** [comment]
> Thanks for reporting this issue. Looks like this is happening for devices having Android 11. Currently, I don't have any device having Android 11 to test this change. I've tried to fix this change and attached APK here 👇🏻.
> 
> [_**Download Sample APK**_](https://github.com/PatilShreyas/EasyUpiPayment-Android/raw/v3.0.3-dev/app/app-debug.apk)
> 
> It's my humble request to test the attached APK and please let me know if it works as expected. If it works, I'll be happy to release the new version `v3.0.3`.

Still facing the same issue, i tested it on MI A3.

**Comment by PatilShreyas** [comment]
Okay, thanks for the quick response.

I think a temporary solution would be declaring package visibility needs from the application side where you're integrating this library. For e.g. you can add package names for UPI apps

```xml
<queries>
    <package android:name="in.amazon.mShop.android.shopping" />
    <package android:name="in.org.npci.upiapp" />
    <package android:name="com.google.android.apps.nbu.paisa.user" />
    <package android:name="com.phonepe.app" />
    <package android:name="net.one97.paytm" />
</queries>
````

For now, you can allow only specific apps for payment instead of choosing among any of them. I'll try to fix this issue ASAP from the library as well.

**Comment by PatilShreyas** [comment]
> > Thanks for reporting this issue. Looks like this is happening for devices having Android 11. Currently, I don't have any device having Android 11 to test this change. I've tried to fix this change and attached APK here 👇🏻.
> > [_**Download Sample APK**_](https://github.com/PatilShreyas/EasyUpiPayment-Android/raw/v3.0.3-dev/app/app-debug.apk)
> > It's my humble request to test the attached APK and please let me know if it works as expected. If it works, I'll be happy to release the new version `v3.0.3`.
> 
> Still facing the same issue, i tested it on MI A3.

Strange, it worked here

https://github.com/PatilShreyas/EasyUpiPayment-Android/issues/112#issuecomment-870460560

**Comment by PatilShreyas** [comment]
Thanks, this has been fixed in the latest release `v3.0.3`.
